### ISSUE TITLE
List state issue while changing config 

### ISSUE THREAD

**Issue by DidarSeyidov** [nanihadesuka/LazyColumnScrollbar#42]
It is not preserve list state when changed screen configuration

**Comment by vhaudiquet** [comment]
@nanihadesuka

I have an issue that might be the same (even this one seems to be cryptic, feel free to move this elsewhere) : 

when I use a listState that can be modified, e.g.

```kotlin
val listState = mutableStateOf(LazyListState())

@Composable
fun List() {
  LazyColumnScrollbar(
            state = listState.value,
            settings = ScrollbarSettings(alwaysShowScrollbar = true)
        ) {
            LazyColumn(
                state = listState.value,
            ) {
                 .....
            }
       }
}
```

then changing the listState (for a new one, with `listState.value = LazyListState()`) makes the scrollbar completely buggy 
(unusable and showing randomly either position 0 or old liststate position).

without `(alwaysShowScrollbar = true)`, the scrollbar just disappears completely

I think the issue is that LazyColumnScrollbar is not recomposed correctly, but LazyColumn is recomposed, so the states are desynchronized ? I could be wrong tho

**Comment by nanihadesuka** [comment]
Yes, it doesn't expect the State object itself to change. Will need a fix
